integration_tests/src/main/python/map_test.py

```diff
@@ -753,8 +753,7 @@ def test_sql_map_scalars(query):
             lambda spark: spark.sql('SELECT {}'.format(query)))
 
 
-@disable_ansi_mode  # ANSI mode failures are tested separately.
-@pytest.mark.parametrize('data_gen', map_gens_sample + [MapGen(IntegerGen(nullable=False, min_val=-100, max_val=100), IntegerGen())], ids=idfn)
+@pytest.mark.parametrize('data_gen', map_gens_sample, ids=idfn)
```

Comment:
map_gens_sample is not a good choice here. It works if we want to show that we don't accidentally mark keys as equal when they are not, but it does not work if we want to show that we do find equal keys. The number of keys that are produced by these is just way too high. We need at least a few that will have a high probability that the same key will show up in both a and b